pkg/operator/baremetal_config.go

```diff
@@ -56,14 +65,13 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		glog.Errorf("provisioningIP not found in Baremetal provisioning CR %s", configName)
 		return BaremetalProvisioningConfig{}, err
 	}
-	provisioningNetworkCIDR, found, err := unstructured.NestedString(provisioningSpec, "provisioningNetworkCIDR")
-	if !found || err != nil {
-		glog.Errorf("provisioningNetworkCIDR not found in Baremetal provisioning CR %s", configName)
-		return BaremetalProvisioningConfig{}, err
+	provisioningDHCPExternal, foundDHCP, err := unstructured.NestedBool(provisioningSpec, "provisioningDHCPExternal")
```

Comment:
> I don't know if that was meant to be interpreted as literally writing back to the resource or not.

If we ever plan to remove it, I would assume we'd have to write back to the resource.

Reply:
I agree with Doug here.  If we're deprecating a field, we should only concern ourselves with that field if the new field is absent.  Also, IMO, the old field and new field should map to some common state here as soon as they're checked, and not referenced directly later (if at all possible).

For example:
```
var newField
if (CR.newField is present) {
  newField = CR.newField
} else {
  // Validate some old fields here first
  ...
  //  Set state of new var
  if CR.oldField == X && CR.oldField2 == y {
    newField == someVal1
  }  else {
    newField == someVal2
}
```

And the reason we want to do this is the logic of translating state between old field and new field is now all handled in one place.  It will make the code elsewhere easier to reason about.

Reply:
Yes, I had thought about the deprecated value differently. I have re-worked my code to always set provisioningNetwork irrespective of the CR containing provisiongDHCPExternal or not. PTAL at the latest version.